Source/WebCore/html/HTMLTemplateElement.cpp

```diff
@@ -68,13 +69,26 @@ DocumentFragment* HTMLTemplateElement::contentIfAvailable() const
     return m_content.get();
 }
 
+DocumentFragment& HTMLTemplateElement::fragmentForInsertion() const
+{
+    if (m_declarativeShadowRoot)
+        return *m_declarativeShadowRoot;
+    return content();
+}
+
 DocumentFragment& HTMLTemplateElement::content() const
 {
+    ASSERT(!m_declarativeShadowRoot);
     if (!m_content)
         m_content = TemplateContentDocumentFragment::create(document().ensureTemplateDocument(), *this);
     return *m_content;
 }
 
+void HTMLTemplateElement::setDeclarativeShadowRoot(ShadowRoot& shadowRoot)
+{
+    m_declarativeShadowRoot = WeakPtr { shadowRoot };
```

Comment:
nit: Do we really need the `WeakPtr { }`?

Reply:
No needed after all. Will fix.